concordium-std-derive/src/lib.rs

```diff
@@ -179,7 +181,11 @@ fn init_worker(attr: TokenStream, item: TokenStream) -> syn::Result<TokenStream>
 
     let fn_name = &ast.sig.ident;
     let rust_export_fn_name = format_ident!("export_{}", fn_name);
-    let wasm_export_fn_name = format!("init_{}", contract_name);
+    let wasm_export_fn_name = format!("init_{}", contract_name.value());
+
+    ContractName::is_valid_contract_name(&wasm_export_fn_name)
+        .map_err(|e| syn::Error::new(contract_name.span(), e.to_string()))?;
```

Comment:
I think this would be easier to understand if you used an `if`, e.g., 
```rust
if let Err(e) = ContractName::is_valid... {
  return syn::Error::new(contract_name.span(), e.to_string())
}
```